workloads/files/workload-network-script-cm.yml

```diff
@@ -0,0 +1,961 @@
+    eval "echo \"${pbench_ssh_template}\"" >> /tmp/pbench-ssh.yaml
+
+    # Loop over pair count
+    for pair_count in ${NETWORK_TEST_PAIRS//,/ }
+    do
+      echo "$(date -u) Running Network benchmarks with ${pair_count} pairs"
+      config_prefix="P2P"
+      network_yaml_template="$(cat /root/workload/p2p-network.yaml.template)"
+      if [ "${NETWORK_TEST_SERVICE}" == "true" ]; then
+        config_prefix="S2S"
+        network_yaml_template="$(cat /root/workload/s2s-network.yaml.template)"
+      fi
+      if [[ "${NETWORK_TEST_HOSTNETWORK}" == "true" ]]; then
+        config_prefix="P2P_HN"
+        oc replace --force -f /tmp/network-scc.yaml
+      fi
+      eval "echo \"${network_yaml_template}\"" >> /tmp/network.yaml
+
+      VIPERCONFIG=/tmp/network.yaml openshift-tests run-test "[Feature:Performance][Serial][Slow] Load cluster should load the cluster [Suite:openshift]"
+
+      # Check if clients and servers are ready
+      for rc_num in `seq 0 $((${pair_count} - 1))`
+      do
+        retries=0
+        while [ $retries -le 60 ] ; do
+          client_ready=`oc get replicationcontroller/uperf-client-${rc_num} -o json -n network0 | jq '.status.replicas==.status.availableReplicas'`
+          if [ "${client_ready}" == "true" ]; then
+            echo "$(date -u) uperf-client-${rc_num} is ready"
+            break
+          else
+            echo "$(date -u) Waiting for uperf-client-${rc_num} to be ready"
+            sleep 1
+          fi
+          retries=$[$retries + 1]
+        done
+        [ "${client_ready}" == "false" ] && echo "$(date -u) uperf-client-${rc_num} failed to be ready on time" && exit 1
+        retries=0
+        while [ $retries -le 60 ] ; do
+          server_ready=`oc get replicationcontroller/uperf-server-${rc_num} -o json -n network0 | jq '.status.replicas==.status.availableReplicas'`
+          if [ "${server_ready}" == "true" ]; then
+            echo "$(date -u) uperf-server-${rc_num} is ready"
+            break
+          else
+            echo "$(date -u) Waiting for uperf-server-${rc_num} to be ready"
+            sleep 1
+          fi
+          retries=$[$retries + 1]
+        done
+        [ "${server_ready}" == "false" ] && echo "$(date -u) uperf-server-${rc_num} failed to be ready on time" && exit 1
+      done
+
+      if [ "${NETWORK_TEST_SERVICE}" == "true" ]; then
+        clients=`oc get svc --output=json -n network0 --selector="name=client" | jq -r '[.items[].spec.clusterIP] | join(",")'`
+        servers=`oc get svc --output=json -n network0 --selector="name=server" | jq -r '[.items[].spec.clusterIP] | join(",")'`
+      else
+        clients=`oc get pods --output=json -n network0 --selector="name=client" | jq -r '[.items[].status.podIP] | join(",")'`
+        servers=`oc get pods --output=json -n network0 --selector="name=server" | jq -r '[.items[].status.podIP] | join(",")'`
+      fi
+      echo "${clients} ${servers}"
```

Comment:
Can we have better logging? The uninitiated will probably not know which is which.

Reply:
Ah leftover debugging, I will remove this.